### ISSUE TITLE
Add "Open Windows Terminal" item to File Explorer backstage menu

### ISSUE THREAD

**Issue by chucker** [microsoft/terminal#5043]
# Description of the new feature/enhancement

File Explorer since Windows 8 uses a ribbon, with a "backstage" File button in the top left. This opens a menu includes a **Open Windows PowerShell** item that will conveniently open PS in the current directory. That seems to be hard-coded on File Explorer's end, and it would be nice to hook into it to use WT instead.

# Proposed technical implementation details (optional)

There might be "Open…" hooks in the registry that the File menu takes advantage of, but I'm not aware of them.

# Duplicate notes

This is not the same issue as #1060, because this is not about context menus.

This is not the same issue as #4109, as that doesn't use the same UI (File --> Open…).

**Comment by zadjii-msft** [comment]
Thanks for calling out the fact this isn't a dupe of those other ones ☺️ 

This does seem like a reasonable request to make on the explorer.exe team. I'm not sure how possible this would actually be. Presumably those links are hardcoded by explorer and apps can't just plug into them.

Maybe there's a way for them to check if we're installed, and hook up buttons in that case.

This is a really good idea. @cinnamon-msft there's some other things you've been cooking up that sound like this. This seems like a _really good idea_ with lots of precedence, can you drive this with the explorer team?

**Comment by DHowett-MSFT** [comment]
Yeah, this is a great thing for us to pursue once v1 is signed off: Kayla and I will drive the discussion with the classic shell folks. Yanking triage and marking Tracking-External.

**Comment by itshoro** [comment]
> This does seem like a reasonable request to make on the explorer.exe team. I'm not sure how possible this would actually be. Presumably those links are hardcoded by explorer and apps can't just plug into them.

Actually you can edit them easily using the registry, see this SO question https://stackoverflow.com/questions/20449316/how-add-context-menu-item-to-windows-explorer-for-folders hence it could be added using administrator privileges.

**Comment by chucker** [comment]
> Actually you can edit them easily using the registry, see this [SO question](https://stackoverflow.com/questions/20449316/how-add-context-menu-item-to-windows-explorer-for-folders) hence it could be added using administrator privileges.

Yes, but this particular issue is not about the context menu, but rather the following UI:

![](https://i.stack.imgur.com/swNt6.png)

**Comment by wbpluto** [comment]
Support this!  Please replace the old "Windows PowerShell" to PS7 or Windows Terminal!

**Comment by ekolis** [comment]
Wait how do you bring back the command prompt option? That disappeared months ago and I want it back if we can't have Windows Terminal pop up here!

**Comment by SitamMatt** [comment]
Hi, is there any progress or plans  for this feature ?

**Comment by zadjii-msft** [comment]
Nope. We'll make sure to update this thread when there is. In the meantime, might I recommend the Subscribe button?
![image](https://user-images.githubusercontent.com/18356694/91237459-5cbb0c80-e700-11ea-9347-b9b1ec2813b1.png)
That way you'll be notified of any updates to this thread, without needlessly pinging everyone on this thread ☺️

**Comment by zadjii-msft** [comment]
You know, the backstage menu is gone entirely in Windows 11. I'm gonna close this one out - if the explorer team ever decides to add it back, we'll talk with them about getting it updated with the Terminal. I'd be absolutely floored if they decided to make any UI changes to the Windows 10 version of explorer at this point.
